### PR TITLE
added histograms to frontend

### PR DESCRIPTION
**Pull request by vayu-stanford**
Histograms mainly added through .js and .html files. New
MultipleRowSetResource class added to facilitate querying of multiple
datasets from server

Note that this commit introduces a new dependency on plotly.js, which is
MIT licensed

### CODE REVIEW

File: src/main/java/macrobase/runtime/resources/MultipleRowSetResource.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+
+import java.util.List;
+import java.util.ArrayList;
+
+@Path("/multiple_rows")
```
Comment (pbailis) [inline src/main/java/macrobase/runtime/resources/MultipleRowSetResource.java]: rows/multiple

Comment (pbailis) [inline src/main/java/macrobase/runtime/resources/MultipleRowSetResource.java]: ??

File: src/main/resources/frontend/console.html [hunk 2]
```diff
@@ -160,12 +161,28 @@ <h3 class="panel-title">Results:</h3>
                         </div>
                     </div>
 
+										<div class="panel  panel-default">
+														<div class="panel-heading">Overview Plot</div>
```
Comment (pbailis) [inline src/main/resources/frontend/console.html]: note panel css

**Comment by edgan8** [comment]
This is super cool, I'd love to try this out and understand how it works! 

Can you add a test plan showing us how to replicate your results? Some screenshots so we know if we're getting the right results would be really helpful too.

It would be nice if we can avoid having a massive 1000 line js file in consolecontroller.js as well to make this extensible in the future though I haven't read that part closely to think about the exact refactoring.

File: src/main/resources/frontend/consoleController.js [hunk 3]
```diff
@@ -340,6 +345,346 @@ myApp.controller('analyzeController', ['$scope', '$http', '$window', 'configServ
...
+            data.push({x:$scope.plotPoints[i][0],
+                opacity:0.5, type:'histogram',
+                histnorm:'probability density',
+                name:$scope.plotPoints[i][1],
+                visible:showSeries
+            });
+        }
+
+        var layout = {
+            title: 'Sampled '+ $scope.plotField + ' Probability Density',
```
Comment (pbailis) [inline src/main/resources/frontend/consoleController.js]: Drop "sampled" from both.

File: src/main/resources/frontend/consoleController.js [hunk 4]
```diff
@@ -349,35 +694,72 @@ myApp.controller('exploreController', ['$scope', '$http', 'configService', 'expl
...
+
+        if($scope.plotPoints.length > 0){
+            data.push({x:$scope.plotPoints[1],
+                opacity:0.75,
+                type:'histogram',
+                histnorm:'probability density',
+                name:'selected itemset'});
+        }
+        var layout = {
+            title: 'Sampled '+ $scope.plotField + ' Values',
```
Comment (pbailis) [inline src/main/resources/frontend/consoleController.js]: Can drop "sampled" too.

File: src/main/resources/frontend/consoleController.js [hunk 5]
```diff
@@ -349,35 +694,72 @@ myApp.controller('exploreController', ['$scope', '$http', 'configService', 'expl
...
+            xaxis: {title: 'Values'},
```
Comment (pbailis) [inline src/main/resources/frontend/consoleController.js]: Change title

File: src/main/resources/frontend/consoleController.js (hunk 4 above)
Comment (pbailis) [inline src/main/resources/frontend/consoleController.js]: drop title entirely?

**Comment by pbailis** [comment]
@vayu-stanford: are you waiting on us for anything?

**Comment by vayu-stanford** [comment]
No - I was a bit tied up over the weekend. I'll be making the suggested changes shortly.

**Comment by vayu-stanford** [comment]
I've incorporated the requested changes into my remote branch.

**Comment by pbailis** [comment]
Can you stress test on campaign_expenditures (on `raiders4`)? `employer_name` versus `receipt_amt`?

**Comment by pbailis** [comment]
Any thoughts on testing Dropwizard? (Not necessary, but might be good to look into.)

**Comment by pbailis** [comment]
Can we rename `itemset` to "subgroup" in all user-facing HTML/JS?

**Comment by pbailis** [comment]
Things to add [in the future, from our discussion]
 - [ ] ability to plot log of PDF values instead of raw PDF values
 - [ ] ability to hide "plot" box in explore view

**Comment by pbailis** [comment]
@vayu-stanford Any updates?

**Comment by vayu-stanford** [comment]
Yes - I have incorporated the suggested changes above. I also stress-tested against the campaign dataset on raiders4 and did not observe any issues introduced by these changes.

**Comment by pbailis** [comment]
LGTM!

**Pull request merged**